aws/handlers/appointments_api_req_handler.go

```diff
@@ -17,15 +17,20 @@ func HandleAppointmentsRequest(req events.APIGatewayProxyRequest) (*events.APIGa
 	switch req.HTTPMethod {
 	case "GET":
 		db, err := database.CreateDB()
-
 		if err != nil {
 			return utils.GatewayResponse(http.StatusBadRequest, models.ErrorBody{ErrorMsg: aws.String(err.Error())})
 		}
 
 		repository := repositories.AppointmentsRepository{DB: db}
 		//TO-DO this is work in progress need to add api to get appointments per user/counselor/timeslot
-		appointments, err := repository.FetchAll(nil)
+		counselorId  := req.QueryStringParameters["counselor"]
+		userId  := req.QueryStringParameters["user"]
+		appointments, err := repository.FetchAll(counselorId, userId)
```

Comment:
fetch is based on ids , could not follow why we need to pass entire counselor , user object

Reply:
There are two reasons behind having method signature accept proper counselor and user instead of ids.

-  it add a layer of protection/validation where we don’t invoke creating a appointment without a valid counselor and user.. passing ids doesn’t guarantee that specific counselor really exists.
- it also allows us in future to use different properties of counselor and user while creating an appointment instead of using just Id...

Reply:
I’d also recommend looking at how to reference a record from another table inside a table record within dynamodb...

Reply:
can you point me to an example , i did go through some tutorials on plural site for get , here we are looking for appointments based on two ids , user id and counselor id , so do we need to do two separate queries for counselors and users to check if they exist ? from my knowloedge in nosql we dont need to add this overhead

Reply:
Mentioned below

Reply:
sorry i dont follow your suggestions, this is not relational table so we need to save ids in the appointments table and use right partition sort keys , querying and checking if individual keys is ok but stitching two tables in the query does not make sense they maybe in different partitions , can you pls write the code so i can follow or perhaps suggest a link from existing code on internet

Reply:
are you perhaps referring to appointments table to contain both counselor and user data ? as explained here https://www.trek10.com/blog/dynamodb-single-table-relational-modeling/ ?

Reply:
@prasadpamidi for dynamodb we need to maintain single table, with primary key secondary key and attributes based on nosql design principles , for example for counselor  PK will be counselor_id, SK will be counselor attributes will have counselor details , similary for user for appointments the SK will have composite key based on appointment status and date , attributes will contain patient and counselor details , letme know your thoughts if you agree will change code